business/workloads.go

```diff
@@ -46,14 +47,15 @@ func NewWorkloadService(k8s kubernetes.ClientInterface, prom prometheus.ClientIn
 
 // WorkloadService deals with fetching istio/kubernetes workloads related content and convert to kiali model
 type WorkloadService struct {
-	prom prometheus.ClientInterface
-	k8s  kubernetes.ClientInterface
 	// Careful not to call the workload service from here as that would be a infinite loop.
 	businessLayer *Layer
 	// The global kiali cache. This should be passed into the workload service rather than created inside of it.
-	cache cache.KialiCache
+	cache   cache.KialiCache
+	clients map[string]kubernetes.ClientInterface
 	// The global kiali config.
 	config *config.Config
+	k8s    kubernetes.ClientInterface
```

Comment:
Yes whenever the workload service gets updated to support multicluster, then the individual k8s client can get removed from this service. I had included some of those changes with this PR originally but removed them so that this change would mostly be about updating the cache.